feat: private files (#525)
pingdotgg/uploadthing: docs/src/pages/regions-and-acl.mdx

import { Callout } from "nextra-theme-docs";

## Regions

UploadThing allows you to select where your files are stored by configuring your

> app's region on the UploadThing dashboard. If no configuration is done, your

Reviewer: ```suggestion
app's region on the UploadThing dashboard. If no configuration is set, your
```

files are uploaded to our default region which is currently `AWS us-west-2`.

<Callout type="info">
  Uploading to a region different than the default one requires a paid plan.
</Callout>

### Configuring region

Go to your app's settings on the UploadThing dashboard. Under `Regions and ACL`,
you can select the region you want to upload your files to. Once changed, all

> new file uploads will go to the selected region.

Reviewer: ```suggestion
**new** file uploads will go to the selected region.
```

Reviewer: just to bring extra attention to this for clarity

import Image from "next/image";

<Image
  src="/region-settings.png"
  alt="Region settings"
  width="1546"
  height="998"
  style={{ marginTop: "1rem", borderRadius: "0.5rem" }}
/>

<Callout type="info">
  Changing your app's region will not move your existing files to the new

>   region, it will only affect new files.

Reviewer: ```suggestion
  region, it will only affect new files. For assistance migrating uploaded files to a new region, 
  please reach out in discord, or via email (ut-support@ping.gg)
```

</Callout>

## Access Controls

By default every file uploaded to UploadThing is accessible simply by it's URL

> (`utfs.io/f/FILE_KEY`). Although this is secure enough for many applications

Reviewer: ```suggestion
(`utfs.io/f/FILE_KEY`). Although this is secure enough for many applications,
some applications require a more secure way to store their files.
```

since the file key is a very hard-to-guess random string, some applications
require a more secure way to store their files.

You can configure your app's access control list (ACL) to restrict access to
your files. UploadThing currently supports two different ACLs:

- `public-read`: Files will be accessible simply by it's URL.

> - `private`: Private files will not be accessible simply by it's URL, but

Reviewer: ```suggestion
- `private`: Private files will not be accessible by it's URL, instead
```

  requires a short-lived signed URL to be accessed.

### Configuring ACL

To configure your app's ACL, go to your app's settings on the UploadThing
dashboard. Under `Regions and ACL`, you can select the ACL you want to use as
default. You may also toggle the ability to override the default ACL on a
per-request basis.

> <Callout type="info">

Reviewer: Maybe combine this with the note above about paid plans, and put it at the very top of the doc. That way if people come to the doc, they can immediately disregard or know if they need to pay for this feature.
